sys/net/gnrc/netif/gnrc_netif.c

```diff
@@ -1826,6 +1787,107 @@ static void _send(gnrc_netif_t *netif, gnrc_pktsnip_t *pkt, bool push_back)
         gnrc_pktbuf_release(pkt);
     }
 #endif /* IS_USED(MODULE_GNRC_NETIF_PKTQ) */
+
+    if (gnrc_netif_netdev_new_api(netif)) {
+        /* with new netdev (with confirm_send), the netif remains owner of the
+         * pkt and is in charge of releasing it once TX is completed */
+        gnrc_pktbuf_release_error(pkt, err);
+    }
+}
+
+#if IS_USED(MODULE_NETDEV_NEW_API)
+/**
+ * @brief   Call the confirm_send handler from an event
+ *
+ * @param[in]   evp     pointer to the event
+ */
+static void _event_handler_tx_done(event_t *evp)
+{
+    gnrc_netif_t *netif = container_of(evp, gnrc_netif_t, event_tx_done);
+    int res = netif->dev->driver->confirm_send(netif->dev, NULL);
+    /* after confirm_send() is called, the device is ready to send the next
+     * frame. So clear netif->tx_pkt to signal readiness */
+    gnrc_pktsnip_t *pkt = netif->tx_pkt;
+    netif->tx_pkt = NULL;
+    bool push_back = (netif->flags | GNRC_NETIF_FLAGS_TX_FROM_PKTQUEUE);
```

Comment:
```suggestion
    bool push_back = netif->flags & GNRC_NETIF_FLAGS_TX_FROM_PKTQUEUE;
```

Otherwise this will always be true.